src/unittest/test.cpp

```diff
@@ -233,6 +236,17 @@ bool run_tests()
 		num_total_tests_run += testmod->num_tests_run;
 	}
 
+	rawstream << "Catch test results: " << std::endl;
+	// Catch prints out the results of this invocation.
+	auto num_catch_tests_failed = Catch::Session().run();
```

Comment:
also it seems the output from catch and later stuff by MT is getting interleaved?
might have to manually flush cout/cerr

Reply:
Shouldn't everything in the output buffer be in order, or is Catch printing the results on a different thread causing the interleaving? I hope not because we don't have the thread ID to be able to join it before printing our own stuff. 😒
If everything is synchronous, then the order of the buffer content should be correct regardless of the points at which the buffer is flushed. It was as I expected on my MSVC build, but there could be a race condition if it's asynchronous.

Reply:
Ah, I considered your thoughts more carefully and I think I follow you now. If we are waiting to write data to both stdout and stderr and we flush them in an interleaved manner, then maybe we could also observe this. So flushing the one containing Catch's prints before doing our own printing should fix the issue.

Reply:
what I mean is this:
![grafik](https://github.com/minetest/minetest/assets/1042418/1869a9fc-70d2-4c57-8dec-3c7d8d74514e)

Reply:
Thank you! I'm glad we have CI as a way to check whether it's been fixed (supposing it's not a race condition). For a first step I'll try manually flushing the right stream as you suggested.

Reply:
I still see this on CI except on gcc_7.

Reply:
This seems to be stdout vs. stderr. Catch prints the summary on stdout: You can verify this by piping stderr into the abyss: `./bin/minetest --run-unittests 2>/dev/null`. We print everything to stderr: Run `./bin/minetest --run-unittests 1>/dev/null`.

One hacky way to make this work should thus be to first flush stderr, then let catch print to stdout, then flush stdout, then let us print to stderr again.

Or we could make both print to stdout / stderr.

Reply:
Alright, I figured out how to make Catch write to `rawstream` and took the liberty to commit that.